File: SDKs/googleapis.com/adexchangebuyer/v1.4/go/pkg/models/operations/adexchangebuyerpretargetingconfigpatch.go
package operations

import (
"openapi/pkg/models/shared")

type AdexchangebuyerPretargetingConfigPatchPathParams struct {
    AccountID string `pathParam:"style=simple,explode=false,name=accountId"`
    ConfigID string `pathParam:"style=simple,explode=false,name=configId"`
    
}

type AdexchangebuyerPretargetingConfigPatchQueryParams struct {
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    UserIP *string `queryParam:"style=form,explode=true,name=userIp"`
    
}

type AdexchangebuyerPretargetingConfigPatchSecurity struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type AdexchangebuyerPretargetingConfigPatchRequest struct {
    PathParams AdexchangebuyerPretargetingConfigPatchPathParams 
    QueryParams AdexchangebuyerPretargetingConfigPatchQueryParams 
    Request *shared.PretargetingConfig `request:"mediaType=application/json"`
    Security AdexchangebuyerPretargetingConfigPatchSecurity 
    
}

type AdexchangebuyerPretargetingConfigPatchResponse struct {
    ContentType string 
    PretargetingConfig *shared.PretargetingConfig 
    StatusCode int64 
    
}

File: SDKs/googleapis.com/adexchangebuyer/v1.4/python/src/sdk/models/operations/adexchangebuyer_pretargetingconfig_patch.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from sdk.models import shared


@dataclass
class AdexchangebuyerPretargetingConfigPatchPathParams:
    account_id: str = field(metadata={'path_param': { 'field_name': 'accountId', 'style': 'simple', 'explode': False }})
    config_id: str = field(metadata={'path_param': { 'field_name': 'configId', 'style': 'simple', 'explode': False }})
    

@dataclass
class AdexchangebuyerPretargetingConfigPatchQueryParams:
    alt: Optional[shared.AltEnum] = field(default=None, metadata={'query_param': { 'field_name': 'alt', 'style': 'form', 'explode': True }})
    fields: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'fields', 'style': 'form', 'explode': True }})
    key: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'key', 'style': 'form', 'explode': True }})
    oauth_token: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'oauth_token', 'style': 'form', 'explode': True }})
    pretty_print: Optional[bool] = field(default=None, metadata={'query_param': { 'field_name': 'prettyPrint', 'style': 'form', 'explode': True }})
    quota_user: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'quotaUser', 'style': 'form', 'explode': True }})
    user_ip: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'userIp', 'style': 'form', 'explode': True }})
    

@dataclass
class AdexchangebuyerPretargetingConfigPatchSecurity:
    oauth2: shared.SchemeOauth2 = field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    oauth2c: shared.SchemeOauth2c = field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclass
class AdexchangebuyerPretargetingConfigPatchRequest:
    path_params: AdexchangebuyerPretargetingConfigPatchPathParams = field()
    query_params: AdexchangebuyerPretargetingConfigPatchQueryParams = field()
    security: AdexchangebuyerPretargetingConfigPatchSecurity = field()
    request: Optional[shared.PretargetingConfig] = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class AdexchangebuyerPretargetingConfigPatchResponse:
    content_type: str = field()
    status_code: int = field()
    pretargeting_config: Optional[shared.PretargetingConfig] = field(default=None)
    
